apps/wing-console/console/server/src/utils/test-runner/test-runner.ts

```diff
@@ -0,0 +1,259 @@
+import type { ITestRunnerClient } from "@winglang/sdk/lib/std/test-runner.js";
+import { TraceType } from "@winglang/sdk/lib/std/test-runner.js";
+
+import type { ConsoleLogger } from "../../consoleLogger.js";
+import type { InternalTestResult } from "../../router/test.js";
+import type { Simulator } from "../../wingsdk.js";
+import { formatTraceError } from "../format-wing-error.js";
+
+import { createSimulatorManager } from "./simulator-manager.js";
+import { createTestStateManager } from "./test-state-manager.js";
+
+export type TestStatus =
+  | "success"
+  | "error"
+  | "idle"
+  | "running"
+  | "uninitialized";
+
+export interface TestItem {
+  id: string;
+  label: string;
+  status: TestStatus;
+  datetime: number;
+  time?: number;
+}
+
+export interface TestRunner {
+  // List of all of the tests and their status.
+  listTests(): Array<TestItem>;
+
+  // Status of the test runner as a whole.
+  status(): TestStatus;
+
+  // Run a single test.
+  runTest(testId: string): void;
+
+  // Run all tests.
+  runAllTests(): void;
+
+  // Register a callback to be called when the tests change.
+  onTestsChange(callback: () => void): void;
+
+  // Initialize the test runner.
+  initialize(): void;
+
+  // Stop the test runner.
+  stop(): void;
```

Comment:
I don't think we need these. Initialization should happen automatically when calling any method, and it should also stop automatically after no use.

I can see the use of a `forceStop` method, though, when the server is being closed.